docs/00-manuscripts/ieee/reviews/reviewer2.md

```diff
@@ -35,32 +35,33 @@ Here Shannon Entropy is defined as `H(X)` which itself is defined as
-Feature selection parameters were treated in the same way as model hyperparameters, i.e. they were optimized during the same stage as model hyperparameters.
-This makes it possible to avoid a two-stage optimization procedure as it would be the case when using wrapper methods instead of filters.
-In our believe there is no way to fully denote the influence of the respective feature selection method on the final performance result due to the joint optimization process.
-However, model hyperparameters needs to be trained and by fixing the tuning range of such and only varying the filter method, we to some degree judge the influence of the chosen method.
-In addition we tried making the used benchmark settings more visible in the manuscript by linking Tables VIII (Hyperparameter settings) and VII (best ten benchmark settings overall) more dominantly.
-Also at the end of section 3.3 (Benchmarking design) we describe the full benchmarking setting and how it is composed
+We would like to thank the reviewer for addressing these issues, which encourages us to explain these important technical aspects in more detail.
+
+First, we completely agree that hyperparameters play a critical role in controlling the flexibility of machine-learning techniques, especially in the case of SVM. Since hyperparameters implement penalization and shrinkage, their optimal values will inevitably also depend on the features being selected, or the decision to perform feature selection at all. This is exactly the reason why model hyperparameters and feature selection hyperparameters were tuned jointly. In our manuscript we specifically state this by saying that "the percentage of features was added as a hyperparameter".
+
+This joint tuning is precisely *necessary* to address the second issue raised by the reviewer, the ability to separate hyperparameter from feature selection influences. If we didn't re-tune the (model) hyperparameters for each feature selection setting, we'd be using non-optimal hyperparameter, which would be an unrealistic and unfair disadvantage for feature selection techniques. We are comparing feature selection techniques with each other, and with models without feature selection, under otherwise equal experimental conditions - i.e. optimal hyperparameters for all.
+
+Hyperparameters were optimized using model-based optimization, which is a state-of-the-art tuning approach. We tried making the used benchmark settings more visible in the revised manuscript by linking Tables VIII (Hyperparameter settings) and VII (best ten benchmark settings overall) more dominantly.
+Also at the end of section 3.3 (Benchmarking design) we describe the full benchmarking setting and how it is composed.
+
+Finally, the entire code is made available in an open-access Github repository. This will not only make the present research reproducible, but it will also allow other researchers to apply the same settings to other use cases.
```

Comment:
```suggestion
We would like to thank the reviewer for addressing these issues, which encourages us to explain these important technical aspects in more detail.

First, we completely agree that hyperparameters play a critical role in controlling the flexibility of machine-learning techniques, especially in the case of SVM.
Since hyperparameters implement penalization and shrinkage, their optimal values will inevitably also depend on the features being selected, or the decision to perform feature selection at all.
This is exactly the reason why model hyperparameters and feature selection hyperparameters were tuned jointly.
In our manuscript we specifically state this by saying that "the percentage of features was added as a hyperparameter".

This joint tuning is precisely *necessary* to address the second issue raised by the reviewer, the ability to separate hyperparameter from feature selection influences.
If we didn't re-tune the (model) hyperparameters for each feature selection setting, we'd be using non-optimal hyperparameter, which would be an unrealistic and unfair disadvantage for feature selection techniques.
We are comparing feature selection techniques with each other, and with models without feature selection, under otherwise equal experimental conditions - i.e. optimal hyperparameters for all.

Hyperparameters were optimized using model-based optimization, which is a state-of-the-art tuning approach.
We tried making the used benchmark settings more visible in the revised manuscript by linking Tables VIII (Hyperparameter settings) and VII (best ten benchmark settings overall) more dominantly.
Also at the end of section 3.3 (Benchmarking design) we describe the full benchmarking setting and how it is composed.

Finally, the entire code is made available in an open-access Github repository.
This will not only make the present research reproducible, but it will also allow other researchers to apply the same settings to other use cases.
```

Thanks - sounds way smoother than my take.